nameserver/mocks_test.go

```diff
@@ -75,6 +75,10 @@ func (mz *mockedZoneWithRecords) AddRecord(ident string, name string, ip net.IP)
 	notImplWarn()
 	return nil
 }
+func (mz *mockedZoneWithRecords) DeleteRecords(ident *string, ip *net.IP, name *string) int {
```

Comment:
...whereas I am looking at the signature and wonder whether that method modifies the args I am passing in.